<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>615</width>
    <height>484</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_3">
      <item>
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Server status:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="serverState">
        <property name="maximumSize">
         <size>
          <width>100</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="startServer">
        <property name="text">
         <string>Start</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="stopServer">
        <property name="text">
         <string>Stop</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>1</number>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>Server</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QPlainTextEdit" name="serverOutput">
          <property name="focusPolicy">
           <enum>Qt::WheelFocus</enum>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QPushButton" name="instantBackup">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;This will backup the current state to a ZIP file.&lt;/p&gt;&lt;p&gt;You will be prompted for a location to save the file to.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>Backup to ZIP file</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="scheduleBackup">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;This will request a backup be taken.&lt;/p&gt;&lt;p&gt;The backup will be saved into the configured backups folder.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>Request an automatic backup</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QLabel" name="nextBackupLabel">
            <property name="text">
             <string>Server status:</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QLineEdit" name="commandEdit"/>
          </item>
          <item>
           <widget class="QPushButton" name="sendButton">
            <property name="text">
             <string>Send</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>Options</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <item>
         <widget class="QLabel" name="invalidServerLocationLabel">
          <property name="text">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600; color:#ff0000;&quot;&gt;You must set the server locaton.&lt;/span&gt;&lt;br/&gt;This should be the root of your minecraft server.&lt;/p&gt;&lt;p&gt;You can download the server &lt;a href=&quot;https://www.minecraft.net/en-us/download/server/bedrock/&quot;&gt;&lt;span style=&quot; text-decoration: underline; color:#007af4;&quot;&gt;from here.&lt;/span&gt;&lt;/a&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="textFormat">
           <enum>Qt::RichText</enum>
          </property>
          <property name="openExternalLinks">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <item>
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>Server location:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="serverFolder"/>
          </item>
          <item>
           <widget class="QPushButton" name="selectServerFolder">
            <property name="text">
             <string>Select folder</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox">
          <property name="title">
           <string>Backups</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <item>
            <widget class="QLabel" name="invalidBackupLocationLabel">
             <property name="text">
              <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600; color:#ff0000;&quot;&gt;Invalid backup folder.&lt;/span&gt;&lt;br/&gt;Automatic backups will not function.&lt;br/&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
             </property>
             <property name="textFormat">
              <enum>Qt::RichText</enum>
             </property>
             <property name="margin">
              <number>0</number>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_5">
             <property name="bottomMargin">
              <number>20</number>
             </property>
             <item>
              <widget class="QLabel" name="label_3">
               <property name="text">
                <string>Automatic backup folder:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="backupFolder"/>
             </item>
             <item>
              <widget class="QPushButton" name="selectBackupFolder">
               <property name="text">
                <string>Select folder</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_8">
             <item>
              <widget class="QCheckBox" name="doRegularBackups">
               <property name="text">
                <string>Schedule a backup at least every:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="backupFrequencyLabel">
               <property name="text">
                <string>n hours</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSlider" name="backupFrequencySlider">
               <property name="minimum">
                <number>1</number>
               </property>
               <property name="maximum">
                <number>48</number>
               </property>
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QCheckBox" name="alwaysBackupOnTime">
             <property name="toolTip">
              <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Checking this will ensure that backups happen every &lt;span style=&quot; font-style:italic;&quot;&gt;x&lt;/span&gt; hours regardless.&lt;/p&gt;&lt;p&gt;Otherwise if a backup is triggered for another reason, the countdown will be restarted.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
             </property>
             <property name="text">
              <string>...even if a backup has happened more recently</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="verticalSpacer_3">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <layout class="QGridLayout" name="gridLayout">
             <item row="0" column="1">
              <widget class="QCheckBox" name="backupOnLeave">
               <property name="text">
                <string>Schedule a backup when a player leaves</string>
               </property>
              </widget>
             </item>
             <item row="0" column="0">
              <widget class="QCheckBox" name="backupOnJoin">
               <property name="text">
                <string>Schedule a backup when a player joins</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <spacer name="verticalSpacer_2">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_6">
             <item>
              <widget class="QLabel" name="backupDelayLabel">
               <property name="text">
                <string>n minutes</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSlider" name="backupDelaySlider">
               <property name="minimum">
                <number>1</number>
               </property>
               <property name="maximum">
                <number>240</number>
               </property>
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>615</width>
     <height>20</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>commandEdit</sender>
   <signal>returnPressed()</signal>
   <receiver>sendButton</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>435</x>
     <y>415</y>
    </hint>
    <hint type="destinationlabel">
     <x>593</x>
     <y>414</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>doRegularBackups</sender>
   <signal>toggled(bool)</signal>
   <receiver>backupFrequencySlider</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>116</x>
     <y>274</y>
    </hint>
    <hint type="destinationlabel">
     <x>181</x>
     <y>298</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>doRegularBackups</sender>
   <signal>toggled(bool)</signal>
   <receiver>backupFrequencyLabel</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>93</x>
     <y>274</y>
    </hint>
    <hint type="destinationlabel">
     <x>68</x>
     <y>298</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>doRegularBackups</sender>
   <signal>toggled(bool)</signal>
   <receiver>alwaysBackupOnTime</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>211</x>
     <y>269</y>
    </hint>
    <hint type="destinationlabel">
     <x>253</x>
     <y>263</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
